File: angular/src/app/models/feedback.ts
import { User } from "./user";

export enum ETypeFeedback {
  Reply = 2,
  Comment = 1
}

export class Feedback {
  id!: number;
  createAt!: string | null;
  updateAt!: string | null;
  mark!: string | null;
  userId!: number | null;
  type!: ETypeFeedback | null;
  recipeId!: number | null;
  user!: User | null;
  feedbackId!: number | null;
  replys: Feedback[] | null = [];

  constructor(type: ETypeFeedback) {
    this.type = type;
  }

}

File: DAL/Models/DB/Feedback.cs
using System;
using System.Collections.Generic;

namespace DAL.Models.DB;

public partial class Feedback
{
    public int Id { get; set; }

    public string? Mark { get; set; }

    public int? UserId { get; set; }

    public int? Type { get; set; }

    public int? RecipeId { get; set; }

    public DateTime? CreateAt { get; set; }

    public DateTime? UpdateAt { get; set; }
    public int? FeedbackId { get; set; }

    public virtual Recipe? Recipe { get; set; }

    public virtual User? User { get; set; }
}
